rust/gitxet/tests/integration_tests/test_integration_scenarios.sh

```diff
@@ -0,0 +1,516 @@
+#!/usr/bin/env bash
+set -e
+set -x
+
+SCRIPT_DIR="$( cd -- "$( dirname -- "${BASH_SOURCE[0]:-$0}"; )" &> /dev/null && pwd 2> /dev/null; )";
+. "$SCRIPT_DIR/initialize.sh"
+
+# Allows pushing to this repo.
+git config --global receive.denyCurrentBranch ignore
+git config --global push.autoSetupRemote true
+
+
+# Make sure the filter is configured globally.
+git xet install 
+
+
+base_dir="`pwd`"
+mkdir config_files/
+
+
+new_bare_repo() { 
+    local name="$1"
+    local repo_dir="$base_dir/$name"
+
+    rm -rf "$repo_dir"
+    mkdir "$repo_dir"
+    pushd "$repo_dir"
+
+    git init --bare
+
+    popd
+}
+
+# Use the local versions of the 
+new_repo() {
+    local name="$1"
+    local repo_dir="$base_dir/$name"
+
+    rm -rf "$repo_dir"
+    mkdir "$repo_dir"
+    pushd "$repo_dir"
+
+    git init
+
+
+    popd
+}
+
+create_file_in_repo() {
+    local name="$1"
+    local repo_dir="$base_dir/$name"
+    local file_name="$repo_dir/$2"
+
+    pushd "$repo_dir"
+    create_data_file "$file_name" 100000
+    git add "$file_name"
+    git commit -m "Added file $file_name"
+    popd
+}
+
+init_repo_as_xet () { 
+    local name="$1"
+    local repo_dir="$base_dir/$name"
+    pushd "$repo_dir"
+
+    config_file_name="$base_dir/config_files/xet_config_file_${name}.toml"
+    
+    echo "[upstream]
+    url = \"${repo_dir}\"
+    " > "${config_file_name}" 
+
+    git xet init -m 2 --force --explicit --write-repo-salt --write-gitattributes --xet-config-file="${config_file_name}"
+
+    popd
+}
+
+intermediate_counter=0
+
+new_tmp_bare_repo () { 
+    local tmp_repo="tmp_repo_${intermediate_counter}"
+    ((intermediate_counter++))
+    >&2 new_bare_repo ${tmp_repo}
+    echo ${tmp_repo}
+}
+
+new_tmp_repo () { 
+    local tmp_repo="tmp_repo_${intermediate_counter}"
+    ((intermediate_counter++))
+    >&2 new_repo ${tmp_repo}
+    echo ${tmp_repo}
+}
+
+
+name_conuter=0
+new_name() { 
+    base=$1
+    ((name_counter++))
+    echo "base_${name_counter}"
```

Comment:
`echo "base_${name_counter}" ` doesn't seem to use var `$base`. Is this desired?

Reply:
Good catch